ThreeDeeFrontend/Components/SearchBar.razor.cs

```diff
@@ -3,42 +3,18 @@
 using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Components;
+using ThreeDeeApplication.Enums;
 using ThreeDeeApplication.Models;
+using ThreeDeeFrontend.ViewModels;
 using ThreeDeeInfrastructure.Repositories;
 
 namespace ThreeDeeFrontend.Components;
 
 public partial class SearchBar
 {
-    [Parameter]
-    public EventCallback<List<FileModel>> FilteredFilesHaveChanged { get; set; }
-
-    [Inject]
-    private IFileRepository FileRepository { get; set; }
-    
-    private List<FileModel> _showFiltered = new();
-    private string _selectedValue;
+    [Parameter] 
+    public string SelectedSearchValue { get; set; }
 
-    protected override void OnParametersSet()
-    {
-        _showFiltered = FileRepository.MockData;
-    }
-
-    private async Task<IEnumerable<string>> Search(string searchValue)
-    {
-        IEnumerable<string> filtered;
-        if (string.IsNullOrEmpty(searchValue))
-        {
-            _showFiltered = FileRepository.MockData;
-            filtered = new List<string>();
-        }
-        else
-        {
-            _showFiltered = FileRepository.MockData.Where(x => x.Name
-                .Contains(searchValue, StringComparison.InvariantCultureIgnoreCase)).ToList();
-            filtered = _showFiltered.Select(x => x.Name);
-        }
-        await FilteredFilesHaveChanged.InvokeAsync(_showFiltered);
-        return filtered;
-    }
+    [Parameter]
+    public Func<string, Task<IEnumerable<string>>> Search { get; set; }
```

Comment:
Was ist das für eine Funktion und warum sieht die so komisch aus :D

Reply:
Ist ein sogenannter Delegat. SearchFunc in SearchBar.razor erwartet eine Methode mit Rückgabe. Wenn man die nicht in der Komponente hat, kann man das ja schlecht über nen Callback oder nur mit etwas wilden Eventkonstruktionen bauen. Also definiert man hier den Delegaten und weist den zu. Ne Ebene höher (Index) wird dann dem Delegaten die eigentliche Methode zugewiesen. Ist also quasi nur ne Art Zwischenschritt und auch eher ein Sonderfall, weil die Prop von dem Framework das hier halt so will.
Aber hast schon recht, nen Schönheitspris gewinnen die nicht. ;-)